### PR TITLE
Add scaleFactor to CLI | Add .cancel function

### LINKED ISSUE #18: Add `scaleFactor` options
Mac file sizes are super large due to retina display.

@alexpelan wrote an implementation of `scaleFactor` for the Swift binary here:
https://github.com/wulkano/Aperture/pull/73

These options will need to make it into `index.ts` on merge:
https://github.com/wulkano/aperture-node/blob/main/index.js#L42

I'm not familiar with Swift, so I'm having trouble building the binary. Tonight I was able to produce `x86_64-apple-macosx/release/Aperture` though this doesn't fit the script found in `package.json` so I'm hesitant to move forward here.

Happy to add the Javascript options if someone can provide the binary.

### PR DESCRIPTION
**Pull request by dac09**
## What does it do?
- Adds scale factor argument to CLI and node function
- Adds new `cancel` function to stop and discard the recording

Fixes #18
Please note I haven't had a chance to test this out yet on this version, but I am using something similar in my older fork

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -61,6 +61,12 @@ declare namespace aperture {
     The `proRes422` and `proRes4444` codecs are uncompressed data. They will create huge files.
     */
     readonly videoCodec?: VideoCodec;
+
+    /**
+     Scale factor to use
+     Useful when recording retina screens, or for resizing the recorded video
```
Comment (sindresorhus) [inline index.d.ts:67]: Needs to document the default value. Use the correct TS doc syntax.

File: index.js [hunk 2]
```diff
@@ -39,14 +39,18 @@ class Aperture {
     macosVersion.assertGreaterThanOrEqualTo('10.13');
   }
 
+
+  recorderTimeout = null
```
Comment (sindresorhus) [inline index.js:43]: Use `undefined`.

File: index.d.ts [hunk 3]
```diff
@@ -101,6 +107,11 @@ declare namespace aperture {
     Returns a `Promise` for the path to the screen recording file.
     */
     stopRecording: () => Promise<string>;
+
+    /**
+     Cancels a recording, if it was started
+     */
+    cancelRecording: () => void;
```
Comment (sindresorhus) [inline index.d.ts:114]: I just noticed that we have been very inconsistent with naming. I think this should be `cancel()`. I'll fix the other methods when this is merged.

File: index.js [hunk 4]
```diff
@@ -227,6 +232,15 @@ class Aperture {
 
     return this.tmpPath;
   }
+
+  cancelRecording() {
+    if (this.recorder !== undefined) {
```
Comment (sindresorhus) [inline index.js:237]: Use early-return.

File: index.js [hunk 5]
```diff
@@ -227,6 +232,15 @@ class Aperture {
...
+      this.recorder.kill();
+      delete this.recorder;
+      delete this.isFileReady;
+      this.recorderTimeout && clearTimeout(this.recorderTimeout)
```
Comment (sindresorhus) [inline index.js:241]: Use `if-statement`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
You need to update the readme too.

**Comment by sindresorhus** [comment]
For next time, the `.cancel()` addition should have been a separate pull request.

File: index.d.ts [hunk 6]
```diff
@@ -61,6 +61,12 @@ declare namespace aperture {
     The `proRes422` and `proRes4444` codecs are uncompressed data. They will create huge files.
     */
     readonly videoCodec?: VideoCodec;
+
+    /**
+     Scale factor to use
+     Useful when recording retina screens, or for resizing the recorded video
+     */
```
[the remark this answers is not in the document]
Comment (dac09) [reply index.d.ts:68]: Is it worth adding prettier to the project? Would avoid these sort of issues.

[the remark this answers is not in the document]
Comment (sindresorhus) [reply index.d.ts:68]: Nah. prettier does indeed make things more consistent, but IMHO also less readable and ugly.

File: index.js [hunk 7]
```diff
@@ -39,14 +39,18 @@ class Aperture {
     macosVersion.assertGreaterThanOrEqualTo('10.13');
   }
 
+
+  recorderTimeout = undefined
```
Comment (sindresorhus) [inline index.js:43]:
```suggestion
  _recorderTimeout = undefined
```

File: index.d.ts [hunk 8]
```diff
@@ -101,6 +108,11 @@ declare namespace aperture {
     Returns a `Promise` for the path to the screen recording file.
     */
     stopRecording: () => Promise<string>;
+
+    /**
+    Cancels a recording, if it was started
```
Comment (sindresorhus) [inline index.d.ts:113]:
```suggestion
    Cancels a recording if it was running.
```

**Comment by sindresorhus** [comment]
Readme and index.d.ts should be fully in sync.

File: index.d.ts [hunk 9]
```diff
@@ -61,6 +61,13 @@ declare namespace aperture {
     The `proRes422` and `proRes4444` codecs are uncompressed data. They will create huge files.
     */
     readonly videoCodec?: VideoCodec;
+
+    /**
+    Scale factor to use
+    The actual height and width of the capture will be multiplied by this number to create the height and width of the output.
+    @default 1
```
Comment (sindresorhus) [inline index.d.ts:68]:
```suggestion
    The scale factor to use.

    The actual height and width of the capture will be multiplied by this number to create the height and width of the output.

    @default 1
```

**Pull request closed without merging**